File: lib/models/test_result.dart
class TestResult {
  final String testName;
  final int executionTimeMs;
  final String details;
  final bool success;

  TestResult(this.testName, this.executionTimeMs, this.details, this.success);

  Map<String, dynamic> toJson() => {
        'testName': testName,
        'executionTimeMs': executionTimeMs,
        'details': details,
        'success': success,
      };

  @override
  String toString() => '$testName: ${executionTimeMs}ms ($details)';
}

File: java/app/src/main/java/com/jossy/android/mobilebenchmarkappjava/models/TestResult.java
package com.jossy.android.mobilebenchmarkappjava.data;

import java.io.Serializable;

public class TestResult implements Serializable {
    private String testName;
    private long executionTimeMs;
    private String details;
    private boolean success;

    public TestResult(String testName, long executionTimeMs, String details, boolean success) {
        this.testName = testName;
        this.executionTimeMs = executionTimeMs;
        this.details = details;
        this.success = success;
    }

    public String getTestName() {
        return testName;
    }

    public long getExecutionTimeMs() {
        return executionTimeMs;
    }

    public String getDetails() {
        return details;
    }

    public boolean isSuccess() {
        return success;
    }
}
